go/obsclient/authclient.go

```diff
@@ -0,0 +1,79 @@
+package obsclient
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/obscuronet/go-obscuro/go/wallet"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/go-obscuro/go/rpcclientlib"
+)
+
+// AuthObsClient extends the functionality of the ObsClient for all methods that require encryption when communicating with the enclave
+// It is created with an EncRPCClient rather than basic RPC client so encryption/decryption is supported
+//
+// The methods in this client are analogous to the methods in geth's EthClient and should behave the same unless noted otherwise.
+type AuthObsClient struct {
+	ObsClient
+}
+
+// NewAuthObsClient constructs an AuthObsClient for sensitive communication with an enclave.
+//
+// It requires an EncRPCClient specifically even though the AuthObsClient uses a Client interface in its struct because
+// the Client interface makes testing easy but an EncRPCClient is required for the actual encrypted communication
+func NewAuthObsClient(client *rpcclientlib.EncRPCClient) *AuthObsClient {
```

Comment:
@BedrockSquirrel Would it be better to hide the `EncRPCClient` from the user, and just take an address and a viewing key? What's the benefit of forcing the construction of the RPC client on the user?